### DISCUSSION TITLE
Mstump causes segfault iff pytorch was imported

### DISCUSSION THREAD

**Discussion by CharlesHolbrow** [TDAmeritrade/stumpy#571] [Help: Other Questions]
I noticed that `stumpy.mstump(...)` consistently causes the Python kernel to crash with `Segmentation fault: 11`. This only happens when I have imported pytorch – `stumpy.mstump(...)` works fine if I do not `import torch`.

Is this a known issue, and could there be a workaround? Does anyone have any troubleshooting tips? Many thanks for any advice/input!

Here's a minimal example:
```python
import numpy as np
import stumpy
import torch
data = np.array([[1, 2, 3, 4, 5, 4], [1, 2, 1, 2, 1, 2]], dtype=np.float64)
mp = stumpy.mstump(data, 3)
```

My suspicion is that there is some incompatibilty with pytorch/Numba. But this is really just a guess.

Here are some specs...
- Running on a 2019 Macbook Pro (This is not one of the new M1 chips)
- Python version 3.7.11
- pytorch version 1.11.0

**Comment by seanlaw** [comment]
@CharlesHolbrow Welcome to the STUMPY community and thank you for your question. So, STUMPY only has 3 dependencies:

1. NumPy
2. SciPy
3. Numba

Can you tell me what versions of these packages are currently installed? I've never tried using STUMPY alongside `pytorch` but I can try to take a look.

**Comment by CharlesHolbrow** [comment]
Thanks, @seanlaw!

1. numpy - `1.20.3`
1. scipy - `1.7.3`
1. numba - `0.51.2`

**Reply by seanlaw** [reply]
I should preface this by saying that I am not a `pytorch` user and this is the first time that we've heard of issues importing `pytorch` along with STUMPY. I started with a brand new `miniconda` installation on my Intel MPB and tried:

```
conda install -c pytorch pytorch torchvision
conda install -c conda-forge -y stumpy pytorch
# numba 0.55.1
# numpy 1.21.2
# python 3.9.7
# pytorch 1.11.0
# scipy 1.7.3
# stumpy 1.10.2
# torchvision 0.12.0
```
Then, I executed your original test case above and was able to print out the multi-dimensional matrix profile without any segmentation faults. Then, I tried the same thing (a fresh install of miniconda) and installed the packages via pip instead:

```
python -m pip install stumpy torch torchvision stumpy
# numba 0.55.1
# numpy 1.21.5
# python 3.9.7
# torch 1.11.0
# scipy 1.8.0
# stumpy 1.10.2
# torchvision 0.12.0
```

And, again, the matrix profile was computed without any segmentation faults. 

@CharlesHolbrow Unfortunately, I am unable to reproduce the same negative behavior as you are observing. Given STUMPY's loose requirements, I'm not certain that this is an issue with STUMPY because, in general, it's been written to be pretty backwards compatible along with having a small dependency footprint.

**Reply by CharlesHolbrow** [reply]
@seanlaw Many thanks for looking into it! I will update this discussion if I can further isolate the issue.

**Reply by seanlaw** [reply]
@CharlesHolbrow Good luck and please report back with any findings as it may help others. Also, I'm looking forward to learning more about how you plan to leverage STUMPY and matrix profiles!

**Reply by seanlaw** [reply]
@CharlesHolbrow I just wanted to follow up and see how things were going?
